File: app/src/main/java/io/celox/xclip/data/ClipboardEntity.java
package io.celox.xclip.data;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "clipboard_history")
public class ClipboardEntity {

    @PrimaryKey(autoGenerate = true)
    private long id;

    private String text;

    private long timestamp;

    public ClipboardEntity(String text, long timestamp) {
        this.text = text;
        this.timestamp = timestamp;
    }

    // Getters and Setters
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/io/celox/xclip/data/ClipboardEntity.kt
package io.celox.xclip.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "clipboard_history")
data class ClipboardEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val text: String,
    val timestamp: Long,
    @ColumnInfo(name = "is_pinned", defaultValue = "0")
    val isPinned: Boolean = false
)
